core/src/main/java/com/radixdlt/bootstrap/GenesisStore.java

```diff
@@ -62,15 +62,13 @@
  * permissions under this License.
  */
 
-package com.radixdlt.genesis.olympia;
+package com.radixdlt.bootstrap;
 
 import com.radixdlt.genesis.GenesisData;
-import com.radixdlt.genesis.olympia.state.OlympiaStateIR;
+import java.util.Optional;
 
-public final class OlympiaStateToBabylonGenesisMapper {
+public interface GenesisStore {
+  void saveGenesisData(GenesisData genesisData);
 
-  public static GenesisData toGenesisData(OlympiaStateIR olympiaStateIR) {
-    // TODO(genesis): coming in a separate PR
-    return GenesisData.testing_default_empty();
-  }
+  Optional<GenesisData> readGenesisData();
```

Comment:
I wonder if we should avoid SBOR decoding the genesis data on bootstrap (to save time) and just hash it directly?

Reply:
I think as a general rule we should only SBOR decode / pass around the full GenesisData where we need to.

Where we can get away with it, just use the hash, and otherwise, just use the raw bytes.

Reply:
:+1: I've refactored it slightly, we no longer hash/sbor when not necessary